quickjs/__init__.py

```diff
@@ -15,15 +17,33 @@ def test():
 
 
 class Function:
-    def __init__(self, name: str, code: str) -> None:
-        self._context = Context()
-        self._context.eval(code)
-        self._f = self._context.get(name)
+    # There are unit tests demonstrating that we are crashing if different threads are accessing the
+    # same runtime, even if it is not at the same time. So we run everything on the same thread in
+    # order to prevent this.
+    _threadpool = concurrent.futures.ThreadPoolExecutor(max_workers=1)
+    
+    def __init__(self, name: str, code: str, *, own_executor=False) -> None:
+        """
+        Arguments:
+            name: The name of the function in the provided code that will be executed.
+            code: The source code of the function and possibly helper functions, classes, global
+                  variables etc.
+            own_executor: Create an executor specifically for this function. The default is False in
+                          order to save system resources if a large number of functions are created.
+        """
+        if own_executor:
+            self._threadpool = concurrent.futures.ThreadPoolExecutor(max_workers=1)
```

Comment:
This only replaces the threadpool of this instance, right?

Reply:
Yes, that's right.